File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    const splittedS = s.split(' ');
    for (const [i, word] of splittedS.entries()) {
        splittedS[i] = [...word].map((v, i) => i % 2 === 0 ? v.toUpperCase() : v.toLowerCase()).join('');
    }
    return splittedS.join(' ');
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        int index = 0;
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (c == ' ') {
                index = 0;
                sb.append(c);
            } else if (index++ % 2 == 0) {
                sb.append(Character.toUpperCase(c));
            } else {
                sb.append(Character.toLowerCase(c));
            } 
        }
        return sb.toString();
    }
}